server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -221,15 +223,22 @@ public Directory directory() {
     public SegmentInfos readLastCommittedSegmentsInfo() throws IOException {
```

Comment:
Not quite following why this update is required (and the additional `readAnySegmentsInfo `) when it doesn't get invoked from ReadOnlyEngine?